core/che-core-api-core/src/main/java/org/eclipse/che/api/core/util/CompositeLineConsumer.java

```diff
@@ -14,38 +14,58 @@
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.nio.channels.ClosedByInterruptException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
 
 /**
  * @author andrew00x
+ * @author Mykola Morhun
  */
 public class CompositeLineConsumer implements LineConsumer {
     private static final Logger LOG = LoggerFactory.getLogger(CompositeLineConsumer.class);
 
-    private final LineConsumer[] lineConsumers;
+    private final List<LineConsumer> lineConsumers;
+    private volatile boolean isClosed;
 
     public CompositeLineConsumer(LineConsumer... lineConsumers) {
-        this.lineConsumers = lineConsumers;
+        this.lineConsumers = new CopyOnWriteArrayList<>(lineConsumers);
+        this.isClosed = false;
     }
 
     @Override
     public void close() throws IOException {
-        for (LineConsumer lineConsumer : lineConsumers) {
-            try {
-                lineConsumer.close();
-            } catch (IOException e) {
-                LOG.error(String.format("An error occurred while closing the line consumer %s", lineConsumer), e);
+        if (!isClosed) {
+            isClosed = true;
+            for (LineConsumer lineConsumer : lineConsumers) {
+                try {
+                    lineConsumer.close();
+                } catch (IOException e) {
+                    LOG.error(String.format("An error occurred while closing the line consumer %s", lineConsumer), e);
```

Comment:
I think that toString in line consumers returns pointer to memory area. Do we need that in error message?